totalRP3/Modules/Register/Characters/RegisterMain.lua

```diff
@@ -681,7 +682,7 @@ local function cleanupProfiles()
 	TRP3_API.Log("Protected profiles: " .. CountTable(protectedProfileIDs));
 	local profilesToPurge = {};
 	for profileID, profile in pairs(profiles) do
-		if not protectedProfileIDs[profileID] and (not profile.time or time() - profile.time > getConfigValue("register_auto_purge_mode")) then
+		if not protectedProfileIDs[profileID] and (TRP3_API.profile.isDefaultProfile(profileID) or not profile.time or time() - profile.time > getConfigValue("register_auto_purge_mode")) then
```

Comment:
Do we have a protection against default profiles ending up in the protected profile table?

Reply:
We don't. I probably should move the default profile check so it ignores protected profiles.
However thinking about it, relations and notes are now invalid to apply to characters with default profile. I don't want to delay the change because I think the current behaviour is unhealthy, but we might have to re-think how to handle those in the future. (Notes implementation in particular is starting to feel very cursed to me.)